tornado_rest_client/test/test_api.py

```diff
@@ -165,6 +175,19 @@ def test_replace_path_tokens(self):
         self.assertEquals(str(ret), '/test/abcd/info')
 
     @testing.gen_test
+    def test_new_style_path_with_property_access(self):
+        test_consumer = RestConsumerTest(client=RestClientTest())
+
+        # the path with an arg must exist as an access method, and fail if you
+        # don't supply the arg.
+        with self.assertRaises(TypeError):
```

Comment:
Why TypeError ? You should intentionally raise some error that describes what went wrong. This seems like falling back to default python failure